test/ambiguity.jl

```diff
@@ -0,0 +1,57 @@
+# ambiguity.jl
+# tests needed only because of code added to resolve method ambiguities
+
+using LinearAlgebra: Diagonal, UpperTriangular
+using LinearAlgebra: Adjoint, Transpose, Symmetric
+using LinearAlgebra: TransposeAbsVec, AdjointAbsVec
+import LinearAlgebra
+using LinearMapsAA
+using Test: @test, @testset
+
+M = rand(3,3)
+A = LinearMapAA(M)
+
+@testset "Diagonal" begin
+    D = Diagonal(1:3)
+    @test Matrix(A * D) == M * D
+    @test Matrix(D * A) == D * M
+end
+
+@testset "UpperTri" begin
+    U = UpperTriangular(M)
+    @test Matrix(A * U) == M * U
+    @test Matrix(U * A) == U * M
+end
+
+@testset "TransposeVec" begin
+    xt = Transpose(1:3)
+    @test xt isa TransposeAbsVec
+    @test xt * M == xt * A
+end
+
+@testset "AdjointVec" begin
+    r = Adjoint(1:3)
+    @test r isa AdjointAbsVec
+    @test r * M == r * A
+end
+
+@testset "Transpose" begin
+# work-around per
+# https://github.com/Jutho/LinearMaps.jl/issues/147
+    LinearAlgebra.isposdef(A::Transpose{Float64, Symmetric{Float64, Matrix{Float64}}}) = false
+    S = LinearAlgebra.Symmetric(M)
+    T = LinearAlgebra.Transpose(S)
```

Comment:
Why do you call `Transpose` here? If you want to test on `Transpose` types, use any plain `A::Matrix` and call `transpose(A)`. That will give you a `Transpose{Float64,Matrix{Float64}}` or similar.

Reply:
Because `transpose` is "too smart" for a `Symmetric` matrix and it does not wrap it with `Transpose` and I need that wrapping to exercise the method that I added to avoid the ambiguity with the strange method discussed over at https://github.com/Jutho/LinearMaps.jl/issues/147